### PR TITLE
gaps: fix gap size in extended

### PR DESCRIPTION
**Pull request by dbungert**
Gap sizing in extended had a flaw when there was a partition after it.  We need the ebr space in between.

Emerged in the private bug LP: #2015521, where a user was given a guided resize option based on a logical partition that was not physically last.  The result was that the end position of the newly created partition was the same spot as the offset of the next partition.

### CODE REVIEW

File: subiquity/common/filesystem/gaps.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def maybe_add_gap(start, end, in_extended):
         if part is None:
             gap_end = ad(device.size - info.min_end_offset)
         else:
-            gap_end = ad(part.offset)
+            gap_end = ad(part.offset - info.ebr_space)
```
Comment (ogayot) [inline subiquity/common/filesystem/gaps.py:153]: It does not feel right to me to subtract `info.ebr_space` if we're not dealing with a logical partition.
Shouldn't it be:
```python
if part.is_logical():
    gap_end = ad(part.offset - info.ebr_space)
else:
    gap_end = ad(part.offset)
```

File: subiquity/tests/api/test_api.py [hunk 2]
```diff
@@ -1469,6 +1469,38 @@ async def test_can_create_unformatted_partition(self):
             v1resp = await inst.get('/storage')
             self.assertEqual([], match(v1resp['config'], type='format'))
 
+    @timeout()
+    async def test_guided_v2_resize_logical_middle_partition(self):
+        '''LP: #2015521 - a logical partition that wasn't the phyiscally last
```
Comment (ogayot) [inline subiquity/tests/api/test_api.py:1474]: minor: physically

File: subiquity/common/filesystem/gaps.py (hunk 1 above)
Comment (dbungert) [reply subiquity/common/filesystem/gaps.py]: Excellent spot.  This would have short-changed primary partitions.  Fixed, and a new unit test.

**Review by ogayot — APPROVED** [review]
LGTM, thanks!

File: subiquity/tests/api/test_api.py (hunk 2 above)
Comment (dbungert) [reply subiquity/tests/api/test_api.py:1474]: I need to just leave the spellchecker on

**Pull request merged**